SharedSources/VLCMediaLibraryManager.swift

```diff
@@ -151,6 +156,52 @@ class VLCMediaLibraryManager: NSObject {
     func genre(sortingCriteria sort: VLCMLSortingCriteria = .default, desc: Bool = false) -> [VLCMLGenre] {
         return medialib.genres(with: sort, desc: desc)
     }
+
+    func fetchMedia(with mrl: URL) -> VLCMLMedia? {
+        return medialib.media(withMrl: mrl)
+    }
+}
+
+// MARK: - Migration
+
+private extension VLCMediaLibraryManager {
+    // Ugly hack to get the realPath
+    func realPath(to url: URL) -> URL? {
+        var path = url.absoluteString
+        path = path.replacingOccurrences(of: "file:///var/mobile", with: "file:///private/var/mobile")
+        return URL(string: path)
+    }
+
+    func migrateMedia() {
+        for media in MLFile.allFiles() as! [MLFile] {
+            if let migratedMedia = fetchMedia(with: realPath(to: media.url)!) {
+                migratedMedia.updateTitle(media.title)
+                migratedMedia.metadata(of: .progress)
+                // FIXME: Need changes in medialibrary, currently no setter for playcount
+                // migratedMedia.playcount = media.playcount.intValue
+                migratedMedia.setMetadataOf(.progress, intValue: media.lastPosition.int64Value)
+            }
+        }
+    }
+
+    func migratePlaylists() {
+        for label in  MLLabel.allLabels() as! [MLLabel] {
+            var mediaForPlaylist = [VLCMLMedia]()
+            for file in label.files as! Set<MLFile> {
+                if let migratedMedia = fetchMedia(with: realPath(to: file.url)!) {
+                    mediaForPlaylist.append(migratedMedia)
+                }
+            }
+            let migratedPlaylist = createPlaylist(with: label.name)
+            mediaForPlaylist.forEach() { migratedPlaylist.appendMedia(withIdentifier: $0.identifier()) }
+        }
+    }
+
+    func migrateToNewMediaLibrary() {
```

Comment:
can you also show the MigrationViewController when you do the migration?

Reply:
Indeed, I completely forgot about `MigrationViewController`!